matrix_client/api.py

```diff
@@ -121,13 +130,18 @@ def register(self, content={}, kind='user'):
                         to complete. "m.login.dummy" is the only non-interactive type.
 
             kind(str): Specify kind="guest" to register as guest.
-        """
+            **query_params: Extra parameters to be sent in the HTTP request.
+                Common values for ``query_params`` are:
 
+                | ts: timestamp for event
+                | user_id: user id for transaction.
+        """
+        query_params['kind'] = kind
         return self._send(
             "POST",
             "/register",
             content=content,
-            query_params={'kind': kind}
+            query_params=query_params
         )
 
     def login(self, login_type, **kwargs):
```

Comment:
This is why I don't like using `**query_params`. Now we've introduced
inconsistency to our api methods. Sometimes, extra kwargs are treated as extra
json, and sometimes they're params.

Additionally, there might be times when an AS that's e2e-aware would make use of
login/logout api methods. Not sure; would need to double-check this.